### PR TITLE
Throttle NFW watson.

### PR DESCRIPTION
**Pull request by heejaechang**
this adds very simple throttle to NFW.

one of issue of NFW compared to fatal watson is that an issue doesn't crash VS which can cause same issue to happen repeatedly within short amount of time.

this can cause the API (PostFault) to attempt to send multiple NFW for same issue in short amount of time causing VS unresponsive.

this is attempt to remove such simple case.

ex) this happened 10+ times in short amount of time

...
Fault bucket , type 0
Event Name: VisualStudioNonFatalErrors2
Response: Not available
Cab Id: 0

Problem signature:
P1: devenv.exe
P2: 15.0.26929.3001
P3: vs.ide.vbcs.nonfatalwatson
P4: System.InvalidOperationException
P5: NuGet.PackageManagement.VisualStudio
P6: NuGet.PackageManagement.VisualStudio.NetCorePackageReferenceProject.GetPackageSpecsAsync
...

I believe this should be handled by PostFault itself (I am in talk with them about it), but until then, I made our side to have simple filtering.

### CODE REVIEW

**Comment by heejaechang** [comment]
@dotnet/roslyn-analysis @dotnet/roslyn-ide can you take a look?

**Comment by heejaechang** [comment]
@dotnet/roslyn-infrastructure ubuntu looks like has an issue in cibuild script. 
https://ci.dot.net/job/dotnet_roslyn/job/master/job/ubuntu_16_debug_prtest/6700/

**Comment by heejaechang** [comment]
retest ubuntu_16_debug_prtest please

File: src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs [hunk 1]
```diff
@@ -22,6 +22,11 @@ internal static class WatsonDisabled
     internal static class WatsonReporter
     {
         /// <summary>
+        /// hold onto last issue we reported
+        /// </summary>
+        private static string s_lastExceptionReported;
```
Comment (mavasani) [inline src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: Can the exception string be very long, if we have lot of exception nesting? Consider using the hash instead.

Comment (heejaechang) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: good idea.

**Comment by heejaechang** [comment]
@Pilchie can I get approval for checking into master?

**Comment by heejaechang** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by Pilchie** [comment]
Approved for 15.5 Preview 2.

**Comment by heejaechang** [comment]
@ivanbasov this is targeting preview 2. so we still can get many NFW for same issue from preview 1.

**Comment by ivanbasov** [comment]
Thanks, @heejaechang! 
@Pilchie, any chance to target this for Preview 1?

**Pull request merged**